cpp/tests/hashing/xxhash64_test.cpp

```diff
@@ -0,0 +1,170 @@
+                                                                    3127544388062992779ul,
+                                                                    1692401401506680154ul,
+                                                                    13770442912356326755ul});
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(output->view(), expected);
+}
+
+TEST_F(HashXX64Test, StringType)
+{
+  auto col1 = cudf::test::strings_column_wrapper(
+    {"The",
+     "quick",
+     "brown fox",
+     "jumps over the lazy dog.",
+     "I am Jack's complete lack of null value",
+     "A very long (greater than 128 bytes/char string) to test a a very long string",
+     "Some multi-byte characters here: ééé",
+     "ééé",
+     "ééé ééé",
+     "ééé ééé ééé ééé",
+     "",
+     "!@#$%^&*(())",
+     "0123456789",
+     "{}|:<>?,./;[]=-"});
+
+  auto output = cudf::hashing::xxhash64(cudf::table_view({col1}));
+
+  // these were generated using the CPU compiled version of the cuco xxhash64 source
+  // https://github.com/NVIDIA/cuCollections/blob/dev/include/cuco/detail/hash_functions/xxhash.cuh
+  auto expected = cudf::test::fixed_width_column_wrapper<uint64_t>({4686269239494003989ul,
+                                                                    6715983472207430822ul,
+                                                                    8148134898123095730ul,
+                                                                    17291005374665645904ul,
+                                                                    2631835514925512071ul,
+                                                                    17518648592268952189ul,
+                                                                    8749004388517322364ul,
+                                                                    17701789113925815768ul,
+                                                                    8612485687958712810ul,
+                                                                    5148645515269989956ul,
+                                                                    17241709254077376921ul,
+                                                                    7379359170906687646ul,
+                                                                    4566581271137380327ul,
+                                                                    17962149534752128981ul});
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(output->view(), expected);
+}
+
+TEST_F(HashXX64Test, TestFixedPoint)
+{
+  auto const col1 = cudf::test::fixed_point_column_wrapper<int32_t>(
+    {0, 100, -100, -999999999, 999999999}, numeric::scale_type{-3});
+  auto const output = cudf::hashing::xxhash64(cudf::table_view({col1}));
+
+  // these were generated using the CPU compiled version of the cuco xxhash64 source
+  // https://github.com/NVIDIA/cuCollections/blob/dev/include/cuco/detail/hash_functions/xxhash.cuh
+  // and passing the 'value' of each input (without the scale) as the decimal-type
```

Comment:
I don't like that cudf hashes decimals the same way regardless of their scale. To me, this is a violation of "different values hash differently". Should we file an issue for discussion here? I think this is a bug but would like a second opinion. My proposal would be to combine the column scale into the hash of the value.

Reply:
> I don't like that cudf hashes decimals the same way regardless of their scale. To me, this is a violation of "different values hash differently". Should we file an issue for discussion here? I think this is a bug but would like a second opinion. My proposal would be to combine the column scale into the hash of the value.

I remember this was discussed when first added and this was the agreed up on implementation. I wasn't part of the discussion so I don't know if it is recorded anywhere. It may be worth creating a separate issue to reopen the discussion.

Reply:
Filed an issue for discussion. I don't recall the original conversation about this, it might have preceded my time. https://github.com/rapidsai/cudf/issues/13705